Homework/Truong_Tran/lesson_6/HouseRobber.java

```diff
@@ -0,0 +1,29 @@
+import java.util.HashMap;
+import java.util.Map;
+
+public class HouseRobber {
+
+    public int rob(TreeNode root) {
+        return robSub(root, new HashMap<>());
+    }
+
+    private int robSub(TreeNode root, Map<TreeNode, Integer> map) {
```

Comment:
in Java, will this `map` value be copied every time you call robSub?

Reply:
in java, it copies address of map variable. So therefore, it will reference to the same object